src/compiler/checker.ts

```diff
@@ -2028,6 +2028,10 @@ namespace ts {
                         let suggestion: Symbol | undefined;
                         if (suggestedNameNotFoundMessage && suggestionCount < maximumSuggestionCount) {
                             suggestion = getSuggestedSymbolForNonexistentSymbol(originalLocation, name, meaning);
+                            // Avoid suggesting "global" if it refers to a global scope augmentation declaration.
+                            if (suggestion && suggestion.valueDeclaration && isAmbientModule(suggestion.valueDeclaration) && isGlobalScopeAugmentation(suggestion.valueDeclaration)) {
+                                suggestion = undefined;
+                            }
                             if (suggestion) {

```

Comment:
Since `getSuggestedLibForNonExistentName` doesn't make sense for `global`, I would just invert the predicate and move it here. Or possibly `return undefined`.

Either way, I think it would be better to extract the predicate to its own variable -- that way you can probably delete the comment on the first line of the change.

Reply:
It's true that `getSuggestedLibForNonExistentName` doesn't make sense for `global`, but execution needs to fall through to the `else` case on line 2049/2053 in order to get any error message at all.

I don't like overwriting `suggestion` in-place either, though. Maybe it'd be better to duplicate the `error` call up here?

```typescript
const isGlobalScopeAugmentationDeclaration = suggestion && suggestion.valueDeclaration && isAmbientModule(suggestion.valueDeclaration) && isGlobalScopeAugmentation(suggestion.valueDeclaration);
if (isGlobalScopeAugmentationDeclaration) {
    if (nameArg) {
        error(errorLocation, nameNotFoundMessage, diagnosticName(nameArg));
    }
}
else if (suggestion) {
    // ...
}
```

Reply:
ooh, I forgot that there weren't any other `error` calls afterward. I guess it's better to keep it the way it is.